### PR TITLE
Catch and handle exception thrown in validation

### PR DESCRIPTION
**Pull request by ncliang**
Hotfix for #inc-rcca-1836-postgres-cdc-source-

### CODE REVIEW

File: debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresConnector.java [hunk 1]
```diff
@@ -100,16 +100,17 @@ public Config validate(Map<String, String> connectorConfigs) {
...
-                }
-                catch (SQLException e) {
-                    logger.info("Failed testing connection for {} with user '{}'", connection.connectionString(),
-                            connection.username());
-                    hostnameValue.addErrorMessage("Unable to connect: " + e.getMessage());
-                }
+                connection.execute("SELECT version()");
+                logger.info("Successfully tested connection for {} with user '{}'", connection.connectionString(),
+                    connection.username());
+            } catch (Throwable e) {
```
Comment (aakashnshah) [inline debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresConnector.java:106]: Why is this changed to `Throwable`?

**Review by aakashnshah — APPROVED** [review]
LGTM, just one question.

File: debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresConnector.java (hunk 1 above)
Comment (ncliang) [reply debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresConnector.java:106]: The exception thrown is a RuntimeException wrapped in a ExecutionException. To be safe, I just made it catch throwable. Exact stack is this:
```
[ERROR] 2020-07-29 05:55:00,853 [qtp702325149-68] org.apache.kafka.connect.runtime.rest.errors.ConnectExceptionMapper toResponse
 - Uncaught exception in REST call to /connector-plugins/io.debezium.connector.postgresql.PostgresConnector/config/validate
java.lang.RuntimeException: Couldn't obtain encoding for database dev
        at io.debezium.connector.postgresql.connection.PostgresConnection.determineDatabaseCharset(PostgresConnection.java:434)
        at io.debezium.connector.postgresql.connection.PostgresConnection.<init>(PostgresConnection.java:77)
        at io.debezium.connector.postgresql.connection.PostgresConnection.<init>(PostgresConnection.java:87)
        at io.debezium.connector.postgresql.PostgresConnector.validate(PostgresConnector.java:102)
        at org.apache.kafka.connect.runtime.AbstractHerder.validateConnectorConfig(AbstractHerder.java:356)
        at org.apache.kafka.connect.runtime.rest.resources.ConnectorPluginsResource.validateConfigs(ConnectorPluginsResource.jav
a:82)
        at jdk.internal.reflect.GeneratedMethodAccessor107.invoke(Unknown Source)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
```

Comment (aakashnshah) [reply debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresConnector.java:106]: I see, do you foresee any situation where catching the exception and swallowing is potentially not good?
